src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -318,19 +343,32 @@ def __init__(self, x, y, curve):
     def public_key(self, backend):
         return backend.load_elliptic_curve_public_numbers(self)
 
-    def encode_point(self):
+    def encode_point(self, compressed=False):
         # key_size is in bits. Convert to bytes and round up
         byte_length = (self.curve.key_size + 7) // 8
-        return (
-            b'\x04' + utils.int_to_bytes(self.x, byte_length) +
-            utils.int_to_bytes(self.y, byte_length)
-        )
+        if not compressed:
+            return (
+                b'\x04' + utils.int_to_bytes(self.x, byte_length) +
+                utils.int_to_bytes(self.y, byte_length)
+            )
+        else:
+            compression_byte = b'\x03' if self.y % 2 else b'\x02'
+            return (
+                compression_byte + utils.int_to_bytes(self.x, byte_length)
+            )
 
     @classmethod
-    def from_encoded_point(cls, curve, data):
+    def from_encoded_point(cls, curve, data, backend=None):
```

Comment:
Let's just deprecate this method rather than updating it to support compressed points. We can raise a `CryptographyDeprecationWarning` and tell consumers to switch to the new method that supports both compressed and uncompressed points. You can see some examples of deprecation in other places in the codebase, but generally you should make a `DeprecatedIn24` in `cryptography.utils` and use `stacklevel=2`.